x-pack/legacy/plugins/watcher/server/np_ready/types.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'src/core/server';
+import { XPackMainPlugin } from '../../../xpack_main/xpack_main';
+import { ElasticsearchPlugin } from '../../../../../../src/legacy/core_plugins/elasticsearch';
+
+export interface ServerShim {
+  route: any;
+  plugins: {
+    xpack_main: XPackMainPlugin;
+    watcher: any;
+    elasticsearch: ElasticsearchPlugin;
+  };
+}
+
+export interface ServerShimWithRouter extends ServerShim {
```

Comment:
>  In the way it's structured now you have to wait till all dependencies are removed before...

Thanks, that makes sense! I'll split them up so that legacy can be cleaned up piecemeal. Likewise regarding second comment.